src/closures/closure_operators.jl

```diff
@@ -0,0 +1,532 @@
+
+    κ⁺¹ = κ(i, j⁺¹, k, grid, closure, u, v, w, T, S)
+    κ⁻¹ = κ(i, j,   k, grid, closure, u, v, w, T, S)
+
+    return (κ⁺¹ * Σ⁺¹ - κ⁻¹ * Σ⁻¹) / grid.Δy
+end
+
+"""
+    ∂y_aca(κ, Σ, i, j, k, grid, closure, u, v, w, T, S)
+
+Compute ``\\partial_y ( \\kappa \\Sigma )``, where `κ` and `Σ`
+are diffusivity and strain functions that return quantities
+at the location `aca`, where `a` is either `f` or `c`.
+The quantity returned has the location `afa`.
+"""
+function ∂y_aca(κ::Function, Σ::Function, i, j, k, grid, closure, u, v, w, T, S)
+    j⁻¹ = decmod1(j, grid.Ny)
+
+    Σ⁺¹ = Σ(i, j,   k, grid, u, v, w)
+    Σ⁻¹ = Σ(i, j⁻¹, k, grid, u, v, w)
+
+    κ⁺¹ = κ(i, j,   k, grid, closure, u, v, w, T, S)
+    κ⁻¹ = κ(i, j⁻¹, k, grid, closure, u, v, w, T, S)
+
+    return (κ⁺¹ * Σ⁺¹ - κ⁻¹ * Σ⁻¹) / grid.Δy
+end
+
+"""
+    ∂z_aaf(κ, Σ, i, j, k, grid, closure, u, v, w, T, S)
+
+Compute ``\\partial_z ( \\kappa \\Sigma )``, where `κ` and `Σ`
+are diffusivity and strain functions that return quantities
+at the location `aaf`, where `a` is either `f` or `c`.
+The quantity returned has the location `aac`.
+"""
+function ∂z_aaf(κ::Function, Σ::Function, i, j, k, grid, closure, u, v, w, T, S)
+    Σ⁻¹ = Σ(i, j, k, grid, u, v, w)
+    κ⁻¹ = κ(i, j, k, grid, closure, u, v, w, T, S)
+
+    if k == grid.Nz
+        return κ⁻¹ * Σ⁻¹ / grid.Δz
+    else
+        Σ⁺¹ = Σ(i, j, k+1, grid, u, v, w)
+        κ⁺¹ = κ(i, j, k+1, grid, closure, u, v, w, T, S)
+
+        # Reversed due to inverse convention for z-indexing
+        return (κ⁻¹ * Σ⁻¹ - κ⁺¹ * Σ⁺¹) / grid.Δz
+    end
+end
+
+"""
+    ∂z_aac(κ, Σ, i, j, k, grid, closure, u, v, w, T, S)
+
+Compute ``\\partial_z ( \\kappa \\Sigma )``, where `κ` and `Σ`
+are diffusivity and strain functions that return quantities
+at the location `aac`, where `a` is either `f` or `c`.
+The quantity returned has the location `aaf`.
+"""
+function ∂z_aac(κ::Function, Σ::Function, i, j, k, grid, closure, u, v, w, T, S)
+    Σ⁺¹ = Σ(i, j, k, grid, u, v, w)
+    κ⁺¹ = κ(i, j, k, grid, closure, u, v, w, T, S)
+
+    if k == 1
+        return - κ⁺¹ * Σ⁺¹ / grid.Δz
+    else
+        Σ⁻¹ = Σ(i, j, k-1, grid, u, v, w)
+        κ⁻¹ = κ(i, j, k-1, grid, closure, u, v, w, T, S)
+
+        # Reversed due to inverse convention for z-indexing
+        return (κ⁻¹ * Σ⁻¹ - κ⁺¹ * Σ⁺¹) / grid.Δz
+    end
+end
+
+# At fcc
+∂x_2ν_Σ₁₁(ijk...) = 2 * ∂x_caa(ν_ccc, Σ₁₁, ijk...)
+∂y_2ν_Σ₁₂(ijk...) = 2 * ∂y_afa(ν_ffc, Σ₁₂, ijk...)
+∂z_2ν_Σ₁₃(ijk...) = 2 * ∂z_aaf(ν_fcf, Σ₁₃, ijk...)
+
+# At cfc
+∂x_2ν_Σ₂₁(ijk...) = 2 * ∂x_faa(ν_ffc, Σ₂₁, ijk...)
+∂y_2ν_Σ₂₂(ijk...) = 2 * ∂y_aca(ν_ccc, Σ₂₂, ijk...)
+∂z_2ν_Σ₂₃(ijk...) = 2 * ∂z_aaf(ν_cff, Σ₂₃, ijk...)
+
+# At ccf
+∂x_2ν_Σ₃₁(ijk...) = 2 * ∂x_faa(ν_fcf, Σ₃₁, ijk...)
+∂y_2ν_Σ₃₂(ijk...) = 2 * ∂y_afa(ν_cff, Σ₃₂, ijk...)
+∂z_2ν_Σ₃₃(ijk...) = 2 * ∂z_aac(ν_ccc, Σ₃₃, ijk...)
+
+"""
+    ∂ⱼ_2ν_Σ₁ⱼ
+
+Return the ``x``-component of the turbulent diffusive flux divergence:
+
+`∂x(2 ν Σ₁₁) + ∂y(2 ν Σ₁₁) + ∂z(2 ν Σ₁₁)`
```

Comment:
Might be good to convert this stuff to LaTeX in a later PR so it renders as math/MathJax in the HTML docs.